### PR TITLE
[OP] Add randperm op

### PR DESCRIPTION
**Pull request by juncaipeng**
Add randperm op:

image
image

### CODE REVIEW

File: python/paddle/tensor/random.py [hunk 1]
```diff
@@ -17,6 +17,108 @@
...
+        out (Variable, optional): Optional output which can be any created 
+            Variable that meets the requirements to store the result of operation.
+            If out is None, a new Varibale will be create to store the result. 
+            Default: None.
+        dtype (np.dtype|core.VarDesc.VarType|str, optional): The type of the 
+            output Tensor. Supported data types: int64, int32. Default: int32.
+        device (str, optional): Specific the output variable to be saved in cpu
+            or gpu memory. Supported None, 'cpu', 'gpu'. If it is None, the output
+            variable will be automatically assigned devices.
+            Default: False.
```
Comment (zhangting2020) [inline python/paddle/tensor/random.py]: Default: False?

File: python/paddle/tensor/random.py [hunk 2]
```diff
@@ -17,6 +17,108 @@
...
+    outputs = {'Out': [out]}
+    attrs = {'n': n, 'dtype': out.dtype, 'seed': seed}
+
+    if device is None:
+        helper.append_op(
+            type='randperm', inputs=inputs, outputs=outputs, attrs=attrs)
+    else:
+        with device_guard(device):
+            helper.append_op(
+                type='randperm', inputs=inputs, outputs=outputs, attrs=attrs)
```
Comment (zhangting2020) [inline python/paddle/tensor/random.py]: The attr(device) supports None in device_guard API, so if statement can be removed.

File: python/paddle/tensor/random.py (hunk 1 above)
Comment (juncaipeng) [reply python/paddle/tensor/random.py]: Done
Default: None.

File: python/paddle/tensor/random.py (hunk 2 above)
Comment (juncaipeng) [reply python/paddle/tensor/random.py]: > The attr(device) supports None in device_guard API, so if statement can be removed.

Done. TKS

File: python/paddle/tensor/random.py [hunk 3]
```diff
@@ -17,6 +17,104 @@
 #            'uniform', 
 #            'shuffle',
 #            'randn',
-#            'randperm',
 #            'rand',
 #            'randint']
+
+from __future__ import print_function
```
Comment (wzzju) [inline python/paddle/tensor/random.py]: Remove the useless package.

File: python/paddle/tensor/__init__.py [hunk 4]
```diff
@@ -169,3 +169,6 @@
 # from .search import index_select   #DEFINE_ALIAS
 # from .search import nonzero   #DEFINE_ALIAS
 # from .search import sort   #DEFINE_ALIAS
+
+__all__ = []
+__all__ += random.__all__
```
Comment (wzzju) [inline python/paddle/tensor/__init__.py]: I don't think these two lines are needed.

File: python/paddle/tensor/random.py [hunk 5]
```diff
@@ -17,6 +17,104 @@
...
+from ..fluid.framework import device_guard
+from ..fluid.layer_helper import LayerHelper
+from ..fluid.data_feeder import convert_dtype, check_variable_and_dtype, check_type, check_dtype
+
+__all__ = ['randperm']
+
+
+def randperm(n,
```
Comment (wzzju) [inline python/paddle/tensor/random.py:33]: 建议使用`@templatedoc()`，参考`clip` Op。

File: python/paddle/fluid/tests/unittests/test_randperm_op.py [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+import unittest
```
Comment (wzzju) [inline python/paddle/fluid/tests/unittests/test_randperm_op.py:17]: You also need a `paddle. randperm ` layer unit test.

File: python/paddle/tensor/random.py [hunk 7]
```diff
@@ -17,6 +17,104 @@
...
+
+    Examples:
+        .. code-block:: python
+	    import paddle
+	    import paddle.fluid as fluid
+
+	    num = 6
+	    is_use_gpu = False
+
+	    data_1 = paddle.tensor.randperm(num)
```
Comment (wzzju) [inline python/paddle/tensor/random.py]: Please use `paddle.randperm ` instead.

File: python/paddle/tensor/random.py (hunk 5 above)
Comment (juncaipeng) [reply python/paddle/tensor/random.py:33]: Done

File: python/paddle/tensor/random.py (hunk 7 above)
Comment (juncaipeng) [reply python/paddle/tensor/random.py]: randperm op is under paddle.tensor

File: paddle/fluid/operators/randperm_op.cc [hunk 8]
```diff
@@ -0,0 +1,98 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserved.
```
Comment (chenwhql) [inline paddle/fluid/operators/randperm_op.cc]: `2016` -> `2020`

File: paddle/fluid/operators/randperm_op.cu [hunk 9]
```diff
@@ -0,0 +1,23 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserved.
```
Comment (chenwhql) [inline paddle/fluid/operators/randperm_op.cu]: `2016` -> `2020`

File: paddle/fluid/operators/randperm_op.h [hunk 10]
```diff
@@ -0,0 +1,65 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserved.
```
Comment (chenwhql) [inline paddle/fluid/operators/randperm_op.h]: `2016` -> `2020`

File: python/paddle/fluid/tests/unittests/test_randperm_op.py [hunk 11]
```diff
@@ -0,0 +1,175 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
```
Comment (chenwhql) [inline python/paddle/fluid/tests/unittests/test_randperm_op.py]: `2018` -> `2020`

Comment (juncaipeng) [reply python/paddle/fluid/tests/unittests/test_randperm_op.py]: Done

File: paddle/fluid/operators/randperm_op.h (hunk 10 above)
Comment (juncaipeng) [reply paddle/fluid/operators/randperm_op.h]: Done tks

**Review by wzzju — APPROVED** [review]
LGTM.

**Review by chenwhql — APPROVED** [review]
LGTM

**Pull request merged**